mcfly/models/model_cnn.py

```diff
@@ -0,0 +1,139 @@
+from tensorflow.keras.models import Sequential
+from tensorflow.keras.layers import Dense, Activation, Convolution1D, \
+    Flatten, BatchNormalization
+from tensorflow.keras.regularizers import l2
+from tensorflow.keras.optimizers import Adam
+import numpy as np
+from argparse import Namespace
+from .base_hyperparameter_generator import generate_base_hyperparameter_set
+
+
+class Model_CNN:
+    """Generate CNN model and hyperparameters.
+    """
+    def __init__(self, x_shape, number_of_classes, metrics=['accuracy'], **settings):
+        """
+
+        Parameters
+        ----------
+        x_shape : tuple
+            Shape of the input dataset: (num_samples, num_timesteps, num_channels)
+        number_of_classes : int
+            Number of classes for classification task
+        metrics : list
+            Metrics to calculate on the validation set.
+            See https://keras.io/metrics/ for possible values.
+        cnn_min_layers : int
+            minimum of Conv layers in CNN model
+        cnn_max_layers : int
+            maximum of Conv layers in CNN model
+        cnn_min_filters : int
+            minimum number of filters per Conv layer in CNN model
+        cnn_max_filters : int
+            maximum number of filters per Conv layer in CNN model
+        cnn_min_fc_nodes : int
+            minimum number of hidden nodes per Dense layer in CNN model
+        cnn_max_fc_nodes : int
+            maximum number of hidden nodes per Dense layer in CNN model
+        """
+        self.model_name = "CNN"
+        self.x_shape = x_shape
+        self.number_of_classes = number_of_classes
+        self.metrics = metrics
+
+        # Set default parameters
+        self.defaults = {
+            'cnn_min_layers': 1,
+            'cnn_max_layers': 10,
+            'cnn_min_filters': 10,
+            'cnn_max_filters': 100,
+            'cnn_min_fc_nodes': 10,
+            'cnn_max_fc_nodes': 2000
+            }
+
+        # Replace default parameters with input
+        for key, value in settings.items():
+            if key in self.defaults:
+                print("The value of {} is set from {} (default) to {}".format(key, self.defaults[key], value))
+
+        # Add missing parameters from default
+        for key, value in self.defaults.items():
+            if key not in settings:
+                settings[key] = value
+        self.settings = settings
```

Comment:
```suggestion
    def __init__(self, x_shape, number_of_classes, metrics=['accuracy'], 'cnn_min_layers'= 1,
            'cnn_max_layers'= 10,
            'cnn_min_filters'= 10,
            'cnn_max_filters'= 100,
            'cnn_min_fc_nodes'= 10,
            'cnn_max_fc_nodes'= 2000, **_other):
        """

        Parameters
        ----------
        x_shape : tuple
            Shape of the input dataset: (num_samples, num_timesteps, num_channels)
        number_of_classes : int
            Number of classes for classification task
        metrics : list
            Metrics to calculate on the validation set.
            See https://keras.io/metrics/ for possible values.
        cnn_min_layers : int
            minimum of Conv layers in CNN model
        cnn_max_layers : int
            maximum of Conv layers in CNN model
        cnn_min_filters : int
            minimum number of filters per Conv layer in CNN model
        cnn_max_filters : int
            maximum number of filters per Conv layer in CNN model
        cnn_min_fc_nodes : int
            minimum number of hidden nodes per Dense layer in CNN model
        cnn_max_fc_nodes : int
            maximum number of hidden nodes per Dense layer in CNN model
        """
        self.model_name = "CNN"
        self.x_shape = x_shape
        self.number_of_classes = number_of_classes
        self.metrics = metrics

        # Set default parameters
        self.settings = {'cnn_min_layers': cnn_min_layers,
                'cnn_max_layers': cnn_max_layers,
                'cnn_min_filters': cnn_min_filters,
                'cnn_max_filters': cnn_max_filters,
                ,'cnn_min_fc_nodes': cnn_min_fc_nodes,
                'cnn_max_fc_nodes', cnn_max_fc_nodes}
```

Reply:
The same could be done for the other model classes.

Reply:
We did something along those lines. However, base_parameters such as learning rate etc. still needed to be added.
(we could look at this again in #253 )